webapp/components/wpt-amend-metadata.js

```diff
@@ -0,0 +1,164 @@
+          margin-bottom: 20px;
+        }
+        .metadataEntry {
+          display: flex;
+          align-items: center;
+        }
+        .link {
+          align-items: center;
+          color: white;
+        }
+      </style>
+      <paper-dialog>
+        <h3>Triage Failing Test</h3>
+        <div class="metadataEntry">
+          <img class="browser" src="[[displayLogo(product)]]">
+          &nbsp; >> [[test]] : &nbsp;
+          <paper-input label="Bug URL" value="{{url}}" autofocus></paper-input>
+        </div>
+        <div class="buttons">
+        <paper-button onclick="[[close]]">Dismiss</paper-button>
+        <paper-button onclick="[[triage]]" dialog-confirm>Triage</paper-button>
+        </div>
+      </paper-dialog>
+      <paper-toast id="showPR" duration="10000"><a id="prLink" class="link" target="_blank" href="[[pr]]"></a></paper-toast>
+`;
+  }
+
+  static get properties() {
+    return {
+      pr: String,
+      url: String,
+      path: String,
+      products: String,
+      test: String,
+      productIndex: Number,
+      product: {
+        type: String,
+        computed: 'computeProduct(productIndex, products)'
+      },
+    };
+  }
+
+  constructor() {
+    super();
+    this.triage = this.handleTriage.bind(this);
+    this.close = this.close.bind(this);
+    this.enter = this.triageOnEnter.bind(this);
+  }
+
+  get dialog() {
+    return this.shadowRoot.querySelector('paper-dialog');
+  }
+
+  open() {
+    this.dialog.open();
+    this.dialog.addEventListener('keydown', this.enter);
+  }
+
+  close() {
+    this.url = '';
+    this.dialog.removeEventListener('keydown', this.enter);
+    this.dialog.close();
+  }
+
+  triageOnEnter(e) {
+    if (e.which === 13) {
+      this.handleTriage();
+      this.close();
+    }
+  }
+
+  computeProduct(productIndex, products) {
+    if (!products) {
+      return;
+    }
+
+    let productVal = [];
+    for (let i = 0; i < products.length; i++) {
+      productVal.push(products[i].browser_name);
+    }
+
+    return productVal[productIndex];
+  }
+
+  getTriagedMetadataMap(product, test) {
+    var link = {};
+    link[test] = [{ 'url': this.url, 'product': product }];
+    return link;
+  }
+
+  displayLogo(product) {
+    if (!product) {
+      return;
+    }
+    return `/static/${product}_64x64.png`;
+  }
+
+  handleTriage() {
+    const url = new URL('/api/metadata/triage', window.location);
+    const fetchOpts = {
+      method: 'PATCH',
+      body: JSON.stringify(this.getTriagedMetadataMap(this.product, this.test)),
+      credentials: 'same-origin',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+    };
+
+    const toast = this.shadowRoot.querySelector('#showPR');
+    const prLink = this.shadowRoot.querySelector('#prLink');
+    this.load(
+      window.fetch(url, fetchOpts)
+        .then(r => {
+          if (!r.ok || r.status !== 200) {
+            this.pr = '';
+            prLink.text = r.status + ': ' + r.statusText;
```

Comment:
Instead of `statusText` (which is empty by default), we want to show the actual body of the response (`r.text()` I think) instead. Try triaging without logging in, users should see "not logged in" in the toast (which is the error message from the API).